All Years/01 - Macros/A01a Set up Universal Macros.sps

```diff
@@ -1,74 +1,170 @@
-﻿* Encoding: UTF-8.
-************************************************************************************************************
-                                                   NSS (ISD)
+* Encoding: UTF-8.
+*Run Macros before SLF update.
 ************************************************************************************************************
 ** AUTHOR:	James McMahon (james.mcmahon@phs.scot)
 ** Date:    	01/08/2018
 ************************************************************************************************************
-** Amended by:         
-** Date:
-** Changes:    
-29/03/19-Localities lookup file link updated -Anita George                 
+** Amended by: Jennifer Thom (Jennifer.Thom@phs.scot)
+** Date: 22/03/21
+** Changes: Switched macro file so there is one universal macro file for each quarterly update. 
+                  *and each year will have its own unique year macro. 
 ************************************************************************************************************.
- * Set the Financial Year.
-Define !FY()
-   "1718"
+*Universal Macros for Source Linkage Files - applies to all years. 
+*Run A01a Set up Universal macros and then A01b Set up (year) Macro at the beginning of each update.
+*Date: 22/03/21. 
+
+*******************************************************.
+ * Update Macros *
+ * Needs changed every update *
+*******************************************************.
+* IT Extracts * 
+* Replace the number with the CSD ref.
+Define !IT_extract_ref()
+    "SCTASK0228343"
+!EndDefine.
+
+*Latest update month for postcode and gp prac lookups.
+Define !LatestUpdate()
+   "May_2021"
 !EndDefine.
 
- * Set the next FY, needed for SPARRA (and HHG).
-Define !NextFY ()
-    "1819"
+Define !Delayed_Discharge_period()
+    "Jul16_Mar21"
+!EndDefine.
+
+
+*******************************************************.
+*Geography Macros.
+*Needs changing when files update.
+*******************************************************.
+
+*Locality file - will need changing when geography files update. 
+Define !Locality_file()
+    "HSCP Localities_DZ11_Lookup_20200825.sav"
+!EndDefine. 
+
+*SPD file - will need changing when geography files update.
+Define !SPD_file()
+    "Scottish_Postcode_Directory_2021_1.sav"
+!EndDefine. 
+
+*SIMD file - will need changing when geography files update.
+Define !SIMD_file()
+    "postcode_2021_1_simd2020v2.sav"
 !EndDefine.
 
-Define !File()
+*DataZone Populations file - will need changing when geography files update.
+Define !DataZone_pop_file()
+    "DataZone2011_pop_est_2011_2019.sav"
+!EndDefine. 
+
+
+*******************************************************.
+* Directories for File Path locations *. 
+*******************************************************.
+*Directory for all years SLF development. Links to !FY() for updating each financial year.
+Define !Year_dir()
    !Quote(!Concat("/conf/sourcedev/Source Linkage File Updates/", !Unquote(!Eval(!FY)), "/"))
```

Comment:
If we change this folder name to get rid of spaces we can remove lots of the quote marks on the various `HOST` commands